diff -up wesnoth-1.17.10/src/gettext.cpp.omv~ wesnoth-1.17.10/src/gettext.cpp
--- wesnoth-1.17.10/src/gettext.cpp.omv~	2022-12-16 02:04:55.280273985 +0100
+++ wesnoth-1.17.10/src/gettext.cpp	2022-12-16 02:14:51.432084097 +0100
@@ -223,8 +223,8 @@ namespace
 			}
 
 			generator_.use_ansi_encoding(false);
-			generator_.categories(bl::message_facet | bl::information_facet | bl::collation_facet | bl::formatting_facet | bl::convert_facet);
-			generator_.characters(bl::char_facet);
+			generator_.categories(bl::category_t::message | bl::category_t::information | bl::category_t::collation | bl::category_t::formatting | bl::category_t::convert);
+			generator_.characters(bl::char_facet_t::char_f);
 			// We cannot have current_locale_ be a non boost-generated locale since it might not supply
 			// the bl::info facet. As soon as we add message paths, update_locale_internal might fail,
 			// for example because of invalid .mo files. So make sure we call it at least once before adding paths/domains
@@ -367,7 +367,7 @@ namespace
 			if(std::has_facet<bl::collator<char>>(current_locale_)) {
 				res << "has bl::collator<char> facet, ";
 			}
-			res << "generator categories='" << generator_.categories() << "'";
+			res << "generator categories='" << static_cast<int>(generator_.categories()) << "'";
 			return res.str();
 		}
 
