diff -up wesnoth-1.19.8/CMakeLists.txt.omv~ wesnoth-1.19.8/CMakeLists.txt
diff -up wesnoth-1.19.8/src/CMakeLists.txt.omv~ wesnoth-1.19.8/src/CMakeLists.txt
--- wesnoth-1.19.8/src/CMakeLists.txt.omv~	2025-01-26 23:45:10.352372293 +0100
+++ wesnoth-1.19.8/src/CMakeLists.txt	2025-01-26 23:55:35.726676593 +0100
@@ -227,6 +227,7 @@ if(ENABLE_GAME)
 		Boost::coroutine
 		Boost::locale
 		Boost::filesystem
+		-lboost_charconv
 		Fontconfig::Fontconfig
 		SDL2::SDL2
 		SDL2::SDL2main
@@ -275,6 +276,7 @@ if(ENABLE_TESTS)
 		Boost::locale
 		Boost::filesystem
 		Boost::unit_test_framework
+		-lboost_charconv
 		Fontconfig::Fontconfig
 		SDL2::SDL2
 		SDL2::SDL2main
@@ -323,6 +325,7 @@ if(ENABLE_SERVER)
 		Boost::coroutine
 		Boost::locale
 		Boost::filesystem
+		-lboost_charconv
 	)
 	if(MSVC)
 		target_link_options(wesnothd PRIVATE /WX)
@@ -364,6 +367,7 @@ if(ENABLE_CAMPAIGN_SERVER)
 		Boost::coroutine
 		Boost::locale
 		Boost::filesystem
+		-lboost_charconv
 	)
 	if(MSVC)
 		target_link_options(campaignd PRIVATE /WX)
