diff -up wesnoth-1.19.10/src/CMakeLists.txt.2~ wesnoth-1.19.10/src/CMakeLists.txt
--- wesnoth-1.19.10/src/CMakeLists.txt.2~	2025-04-12 00:34:08.659477347 +0200
+++ wesnoth-1.19.10/src/CMakeLists.txt	2025-04-12 00:34:08.664764463 +0200
@@ -227,6 +227,7 @@ if(ENABLE_GAME)
 		Boost::locale
 		Boost::filesystem
 		-lboost_charconv
+		boost_process
 		Fontconfig::Fontconfig
 		SDL2::SDL2
 		SDL2::SDL2main
@@ -324,6 +325,7 @@ if(ENABLE_SERVER)
 		Boost::locale
 		Boost::filesystem
 		-lboost_charconv
+		boost_process
 	)
 	if(MSVC)
 		target_link_options(wesnothd PRIVATE /WX)
diff -up wesnoth-1.19.10/src/filesystem.cpp.2~ wesnoth-1.19.10/src/filesystem.cpp
--- wesnoth-1.19.10/src/filesystem.cpp.2~	2025-03-17 05:29:07.000000000 +0100
+++ wesnoth-1.19.10/src/filesystem.cpp	2025-04-12 00:34:08.664913684 +0200
@@ -36,6 +36,8 @@
 #include <boost/iostreams/device/file_descriptor.hpp>
 #include <boost/iostreams/stream.hpp>
 #include <boost/process.hpp>
+#include <boost/process/v1/search_path.hpp>
+#include <boost/algorithm/string/replace.hpp>
 #include "game_config_view.hpp"
 
 #ifdef _WIN32
@@ -1011,14 +1013,14 @@ std::string get_exe_path()
 	// with version
 	std::string version = std::to_string(game_config::wesnoth_version.major_version()) + "." + std::to_string(game_config::wesnoth_version.minor_version());
 	std::string exe = filesystem::get_program_invocation("wesnoth-"+version);
-	bfs::path search = bp::search_path(exe).string();
+	bfs::path search = bp::v1::search_path(exe).string();
 	if(!search.string().empty()) {
 		return search.string();
 	}
 
 	// versionless
 	exe = filesystem::get_program_invocation("wesnoth");
-	search = bp::search_path(exe).string();
+	search = bp::v1::search_path(exe).string();
 	if(!search.string().empty()) {
 		return search.string();
 	}
@@ -1779,7 +1781,7 @@ std::string sanitize_path(const std::str
 
 	std::string canonicalized = filesystem::normalize_path(path, true, false);
 	if(user_name != nullptr) {
-		boost::replace_all(canonicalized, user_name, "USER");
+		boost::algorithm::replace_all(canonicalized, user_name, "USER");
 	}
 
 	return canonicalized;
diff -up wesnoth-1.19.10/src/game_launcher.cpp.2~ wesnoth-1.19.10/src/game_launcher.cpp
--- wesnoth-1.19.10/src/game_launcher.cpp.2~	2025-03-17 05:29:07.000000000 +0100
+++ wesnoth-1.19.10/src/game_launcher.cpp	2025-04-12 00:36:39.441149257 +0200
@@ -55,6 +55,7 @@
 #include <boost/process/windows.hpp>
 #endif
 #include <boost/process.hpp>
+#include <boost/process/v1/child.hpp>
 #include <cstdlib>   // for system
 #include <new>
 #include <thread>
@@ -784,9 +785,9 @@ void game_launcher::start_wesnothd()
 	try
 	{
 #ifndef _WIN32
-		bp::child c(wesnothd_program, "-c", config);
+		bp::v1::child c(wesnothd_program, "-c", config);
 #else
-		bp::child c(wesnothd_program, "-c", config, bp::windows::create_no_window);
+		bp::v1::child c(wesnothd_program, "-c", config, bp::windows::create_no_window);
 #endif
 		c.detach();
 		// Give server a moment to start up
@@ -794,7 +795,7 @@ void game_launcher::start_wesnothd()
 		std::this_thread::sleep_for(50ms);
 		return;
 	}
-	catch(const bp::process_error& e)
+	catch(const bp::v1::process_error& e)
 	{
 		prefs::get().set_mp_server_program_name("");
 
